cachito/workers/pkg_managers/gomod.py

```diff
@@ -253,6 +258,30 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
         return {"module": module, "module_deps": module_level_deps, "packages": packages}
 
 
+def _merge_bundle_dirs(root_src_dir, root_dst_dir):
+    """
+    Merge two bundle directories together.
+
+    The contents of root_src_dir will be copied into root_dst_dir, overwriting any files
+    that might already be present. For a description of the algorithm, see
+    https://lukelogbook.tech/2018/01/25/merging-two-folders-in-python/
```

Comment:
I'm afraid I don't have enough golang experience to know if merging the downloaded directories like this is always correct or not. @athos-ribeiro @lcarva, do you guys know?

Reply:
I do not have the golang experience either, necessarily... but I assumed that this would work since I thought the directory structure would need to take into account the versions of the packages in order to allow for a package requiring multiple versions of a the same dependency.

Reply:
A bit of investigation: https://pastebin.com/iiBrmq99
Each package directory seems to have a `list` file listing the available versions, this may need to be accounted for.

Reply:
Here's repos you can use for testing:
https://github.com/chmeliik/cachito-sample-package/tree/gomod-2-versions-one-module
https://github.com/chmeliik/cachito-sample-package/tree/gomod-2-versions-2-modules

and the corresponding requests:
```shell
curl "localhost:8080/api/v1/requests" \
    -X POST \
    -H 'content-type: application/json' \
    -d '{
          "repo": "https://github.com/chmeliik/cachito-sample-package/",
          "ref": "d48d2ba80f8c5ec9812e55f30422f433dee1d27d",
          "pkg_managers": ["gomod"]
        }'

curl "localhost:8080/api/v1/requests" \
    -X POST \
    -H 'content-type: application/json' \
    -d '{
          "repo": "https://github.com/chmeliik/cachito-sample-package/",
          "ref": "cb9fedf52fe6e6cd09bfe5092d428eeeb27457b3",
          "pkg_managers": ["gomod"],
          "packages": {
            "gomod": [
              {"path": "module_a"},
              {"path": "module_b"}
            ]
          }
        }'
```

Reply:
This should be resolved with the latest commit: https://pastebin.com/diYe9Aje